sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientIntegrationTest.java

```diff
@@ -131,4 +141,50 @@ void peekBatchMessagesFromSequence() {
             .expectNextCount(maxMessages)
             .verifyComplete();
     }
+
+    /**
+     * Verifies that we can renew message lock.
+     */
+    @Test
+    void renewMessageLock() {
+        // Arrange
+        Duration renewAfterSeconds = Duration.ofSeconds(1);
+        long takeTimeToProcessMessageMillis = 10000;
+        final String contents = "Some-contents";
+        final ServiceBusMessage message = TestUtils.getServiceBusMessage(contents, "id-1", 0);
+
+        AtomicReference<Integer> renewMessageLockCounter = new AtomicReference<>(0);
+        // Assert & Act
+        StepVerifier.create(sender.send(message).thenMany(receiverManual.receive()
+            .take(1)
+            .delayElements(renewAfterSeconds)
+            .map(receivedMessage -> {
+                //  keep renewing the lock whole you process the message.
+                Disposable disposable = receiverManual.renewMessageLock(receivedMessage.getLockToken())
```

Comment:
I know you added comments and it doesn't make it reactive nor does it use StepVerifier correctly.
* You've only hooked up the `doOnNext`, which is used for side effects like logging, but you use it for validation. the first overload of `subscribe` is a better fit.
* What if the renewal fails? There is no handling of any error cases to fail the test
* What if the operation never emits a value because it was hung? You don't handle that case either.

Since we are pressed for time, can you create an issue to write tests for this?

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/9193